src/main/java/dev/jbang/cli/BaseBuildCommand.java

```diff
@@ -83,8 +83,17 @@ void setJavaVersion(String javaVersion) {
 	@CommandLine.Option(names = { "--cp", "--class-path" }, description = "Add class path entries.")
 	List<String> classpaths;
 
+	@CommandLine.Option(names = {
+			"-f", "--fresh" }, description = "Force script upate. A new jar is created.", defaultValue = "false")
+	boolean forceUpdate;
+
 	PrintStream out = new PrintStream(new FileOutputStream(FileDescriptor.out));
 
+	protected void deleteJarFile(String scriptOrFile) throws IOException {
+		if (Util.getJarFile(scriptOrFile).delete())
```

Comment:
is this even necessary ? don't it just overwrite the existing one ? 

and if it needs deleting the jar, how about the .jar directory that has the actual built content ?

Reply:
It was needed, but applying @quintesse suggestion all this jar logic is not necessarily anymore. =)